torch/nn/_functions/thnn/upsampling.py

```diff
@@ -62,15 +59,39 @@ def backward(self, grad_output):
         return grad_input
 
 
-class UpsamplingBilinear2d(_UpsamplingBase):
+class UpsamplingBilinear2d(Function):
+
+    def __init__(self, size=None, scale_factor=None):
+        super(UpsamplingBilinear2d, self).__init__()
+        if size is None and scale_factor is None:
+            raise ValueError('either size or scale_factor should be defined')
+        if scale_factor is not None:
+            if not isinstance(scale_factor, (Integral, tuple)):
```

Comment:
I used this constraint for two reasons:

1) the current codebase and THNN both enforce integer scaling factors
2) while you can request non-integer scaling factors implicitly by using the `size` keyword in bilinear upsampling, doing this explicitly for arbitrary floats is not well defined for any resulting image that doesn't have integer pixel sizes.  Does one round up?  Down? 
3) I wanted to conceptually mirror the pooling modules as much as possible.